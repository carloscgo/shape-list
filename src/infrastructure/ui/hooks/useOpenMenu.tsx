// infrastructure/ui/hooks/useOpenMenu.tsx

import { useCallback } from 'react'
import { Some } from '../../../application'

type PropsUse = {
  sidebar: Some
  maxSidebar: Some
  miniSidebar: Some
  maxToolbar: Some
  logo: Some
  content: Some
}

const useOpenMenu = ({
  sidebar,
  maxSidebar,
  miniSidebar,
  maxToolbar,
  logo,
  content,
}: PropsUse) => {
  const openNav = useCallback(() => {
    if (sidebar.current.classList.contains('-translate-x-48')) {
      // max sidebar
      sidebar.current.classList.remove('-translate-x-48')
      sidebar.current.classList.add('translate-x-none')
      maxSidebar.current && maxSidebar.current.classList.remove('hidden')
      maxSidebar.current && maxSidebar.current.classList.add('flex')
      miniSidebar.current && miniSidebar.current.classList.remove('flex')
      miniSidebar.current && miniSidebar.current.classList.add('hidden')
      maxToolbar.current && maxToolbar.current.classList.add('translate-x-0')
      maxToolbar.current &&
        maxToolbar.current.classList.remove('translate-x-24', 'scale-x-0')
      logo.current && logo.current.classList.remove('ml-12')
      content.current && content.current.classList.remove('ml-12')
      content.current && content.current.classList.add('ml-12', 'md:ml-60')
    } else {
      // mini sidebar
      sidebar.current.classList.add('-translate-x-48')
      sidebar.current.classList.remove('translate-x-none')
      maxSidebar.current && maxSidebar.current.classList.add('hidden')
      maxSidebar.current && maxSidebar.current.classList.remove('flex')
      miniSidebar.current && miniSidebar.current.classList.add('flex')
      miniSidebar.current && miniSidebar.current.classList.remove('hidden')
      maxToolbar.current &&
        maxToolbar.current.classList.add('translate-x-24', 'scale-x-0')
      maxToolbar.current && maxToolbar.current.classList.remove('translate-x-0')
      logo.current && logo.current.classList.add('ml-12')
      content.current && content.current.classList.remove('ml-12', 'md:ml-60')
      content.current && content.current.classList.add('ml-12')
    }
  }, [sidebar, maxSidebar, miniSidebar, maxToolbar, logo, content])

  return openNav
}

export default useOpenMenu
