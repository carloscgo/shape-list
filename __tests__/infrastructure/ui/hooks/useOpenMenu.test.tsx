import { useRef } from 'react'
import {
  expect,
  describe,
  it,
  renderHook,
  render,
  fireEvent,
  React,
} from '../../../../setupTest'
import useOpenMenu from '../../../../src/infrastructure/ui/hooks/useOpenMenu'

describe('Use Open Menu Hook', () => {
  const Component = () => {
    const sidebar = useRef<HTMLDivElement>(null)
    const maxSidebar = useRef<HTMLDivElement>(null)
    const miniSidebar = useRef<HTMLDivElement>(null)
    const maxToolbar = useRef<HTMLDivElement>(null)
    const logo = useRef<HTMLDivElement>(null)
    const content = useRef<HTMLDivElement>(null)

    const props = {
      sidebar,
      maxSidebar,
      miniSidebar,
      maxToolbar,
      logo,
      content,
    }

    const openNav = useOpenMenu(props)

    return (
      <div aria-label="openNav" onClick={() => openNav()}>
        <div ref={sidebar}></div>
        <div ref={maxSidebar}></div>
        <div ref={miniSidebar}></div>
        <div ref={maxToolbar}></div>
        <div ref={logo}></div>
        <div ref={content}></div>
      </div>
    )
  }

  it('should toggle classnames when called', () => {
    const { getByLabelText, container } = render(<Component />)

    const openNav = getByLabelText('openNav')

    fireEvent.click(openNav)

    expect(container).toMatchSnapshot()

    fireEvent.click(openNav)

    expect(container).toMatchSnapshot()
  })
})
