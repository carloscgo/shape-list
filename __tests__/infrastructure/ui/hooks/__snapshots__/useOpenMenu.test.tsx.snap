// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Use Open Menu Hook > should toggle classnames when called 1`] = `
<div>
  <div
    aria-label="openNav"
  >
    <div
      class="-translate-x-48"
    />
    <div
      class="hidden"
    />
    <div
      class="flex"
    />
    <div
      class="translate-x-24 scale-x-0"
    />
    <div
      class="ml-12"
    />
    <div
      class="ml-12"
    />
  </div>
</div>
`;

exports[`Use Open Menu Hook > should toggle classnames when called 2`] = `
<div>
  <div
    aria-label="openNav"
  >
    <div
      class="translate-x-none"
    />
    <div
      class="flex"
    />
    <div
      class="hidden"
    />
    <div
      class="translate-x-0"
    />
    <div
      class=""
    />
    <div
      class="ml-12 md:ml-60"
    />
  </div>
</div>
`;
